### PR TITLE
MVC: migrate CSVListField types to more sensible fields where possible

### PR DESCRIPTION
**Pull request by swhite2**
Extend the `MacAddressField` as a list type for usage in Captive Portal. Of note here is the Netflow `MaskPerItem` property, it likely makes more sense to extend either the `HostnameField` or `NetworkField` to allow <ip>:<port> combinations, putting this up for discussion here.

### CODE REVIEW

File: src/opnsense/mvc/app/models/OPNsense/Diagnostics/Netflow.xml [hunk 1]
```diff
@@ -27,6 +27,7 @@
             <targets type="CSVListField">
                 <Mask>/^((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?):(6553[0-5]|655[0-2][0-9]|65[0-4](\d){2}|6[0-4](\d){3}|[1-5](\d){4}|[1-9](\d){0,3})([,]){0,1})*/u</Mask>
```
Comment (fichtner) [inline src/opnsense/mvc/app/models/OPNsense/Diagnostics/Netflow.xml:28]: it requires simplifying the regex which is the original motivation :)

File: src/opnsense/mvc/app/models/OPNsense/CaptivePortal/CaptivePortal.xml [hunk 2]
```diff
@@ -58,13 +58,14 @@
...
+                    <asList>Y</asList>
+                    <WildcardEnabled>N</WildcardEnabled>
                     <ValidationMessage>Please enter valid addresses</ValidationMessage>
                 </allowedAddresses>
-                <allowedMACAddresses type="CSVListField">
-                    <ChangeCase>lower</ChangeCase>
-                    <Mask>/^((([0-9a-fA-F]{2}:){5}([0-9a-fA-F]{2})([,]){0,1}))*/u</Mask>
+                <allowedMACAddresses type="MacAddressField">
+                    <asList>Y</asList>
                     <ValidationMessage>Please enter valid mac addresses</ValidationMessage>
```
Comment (fichtner) [inline src/opnsense/mvc/app/models/OPNsense/CaptivePortal/CaptivePortal.xml:69]: better to drop this then

File: src/opnsense/mvc/app/models/OPNsense/CaptivePortal/CaptivePortal.xml [hunk 3]
```diff
@@ -58,13 +58,14 @@
                     <Mask>/^([0-9a-zA-Z.,_\-]){0,1024}$/u</Mask>
                     <ValidationMessage>Please enter a valid servername, ip address or leave this option blank</ValidationMessage>
                 </servername>
-                <allowedAddresses type="CSVListField">
-                    <Mask>/^([\/0-9.,])*/u</Mask>
+                <allowedAddresses type="NetworkField">
+                    <FieldSeparator>,</FieldSeparator>
+                    <asList>Y</asList>
+                    <WildcardEnabled>N</WildcardEnabled>
                     <ValidationMessage>Please enter valid addresses</ValidationMessage>
```
Comment (fichtner) [inline src/opnsense/mvc/app/models/OPNsense/CaptivePortal/CaptivePortal.xml:65]: drop

File: src/opnsense/mvc/app/models/OPNsense/Diagnostics/Netflow.xml (hunk 1 above)
Comment (swhite2) [reply src/opnsense/mvc/app/models/OPNsense/Diagnostics/Netflow.xml]: Simplified it, not a whole lot shorter. The other option is a custom field type, though this works just fine

**Review by fichtner — APPROVED** [review]
Let's go step by step?

**Comment by swhite2** [comment]
I can add the IP:Port field in a separate PR if you want :)

**Comment by fichtner** [comment]
Yes please. RC1 should have as much as possible included.

**Pull request merged**